main.go

```diff
@@ -238,11 +241,11 @@ type creds struct {
 }
 
 type node struct {
-	Id           string      `json:"id,omitempty"`
-	Cluster      string      `json:"cluster,omitempty"`
-	Metadata     interface{} `json:"metadata,omitempty"`
```

Comment:
Was there any particular reason to have this as an `interface{}` instead of a concrete type. I just changed it to make it easier to directly add to the map in the `v3` case.

Reply:
I don't expect there was any particular reason, other than it was a bit unclear if we'd use a struct or a map.